### PR TITLE
feat: nostr-to-evm eip-712 cert

### PR DESCRIPTION
**Pull request by daopunk**
## Summary

Squad members can now prove which EVM address belongs to their Nostr identity, and peers can verify that binding without trusting unsigned gossip.

Before this change, roster EVM shares on `#announcements` were plain JSON — any member could claim another npub's address, and late joiners had no certified catch-up path. The codebase also still called the Commons join identity "squad bot," which collided with unrelated bot concepts.

After this change, each member signs an off-chain **EIP-712 bind certificate** (`SquadRosterBind`) tying `(parentId, memberNpub)` to their roster EVM address. Shares ship as `squad_member_evm_share` **v2** with `issued_at` + signature; the backend verifies before persisting. Peers can relay certified rows via `squad_evm_roster_snapshot` on state-sync request. The cert is **display and gossip only** — ACL still resolves identity from the `#personal-alerts` binding, not the cert alone (see `docs/governance/ACCESS_CONTROL.md`).

This PR also finishes the **Join inbox** rename (replacing "squad bot" across wire schemas, Rust module, UI, and docs) and fixes several join-inbox lifecycle bugs uncovered during the work.

## Changes

- **EIP-712 roster bind cert** — new `roster_bind_cert` Rust module (`sign_squad_roster_bind_cert` command), golden-vector tests, and DB ingest that rejects unsigned, forged, stale, or author-mismatched certs.
- **Certified MLS wire** — `squad_member_evm_share` v2 and `squad_evm_roster_snapshot` on `#announcements`; frontend publish/parse paths updated.
- **Join inbox rename** — `squad_bot` → `join_inbox` (module, commands, schemas, i18n, UI components); `SQUAD_BOT_JOIN.md` replaced by `docs/communities/JOIN_INBOX.md`.
- **Join inbox fixes** — holder key transfer over DM, virgin-squad initialization gate (creator normalization + error propagation to Settings), broadcast duration label typo, deduplicated warning toasts.
- **UI polish** — squad invite card refactor; dashboard crew/roles copy aligned with Join inbox terminology.

## Test plan

- [ ] `pnpm test` — frontend unit tests (roster bind, join-inbox, evm share/snapshot).
- [ ] `cd src-tauri && cargo test roster_bind_cert` — sign/verify roundtrip, golden digest, stale-cert rejection, forged ingest.
- [ ] Bind squad EVM on `#personal-alerts` → confirm v2 share appears on `#announcements` with signature; Crew tab shows bound address.
- [ ] Second client requests state sync → `squad_evm_roster_snapshot` applies peer certs without author-mismatch.
- [ ] Create squad with Commons → Join inbox initializes; holder can transfer key via DM; Settings Initialize works on virgin squads.
- [ ] Request to join via Commons card → DM reaches Join inbox holders; accept/reject flows in Settings.

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)

### CODE REVIEW

File: src-tauri/src/db.rs [hunk 1]
```diff
@@ -4109,19 +4115,33 @@ pub fn upsert_squad_member_evm<R: Runtime>(
     let norm = crate::evm::normalize_hex_address(evm_address.trim())
         .ok_or_else(|| "Invalid EVM address".to_string())?;
     crate::evm::evm_accounts::ensure_address_allowed_on_squad_roster(&handle, norm.as_str())?;
+    let issued = issued_at.unwrap_or(0);
+    let sig = bind_signature.unwrap_or_default();
+    if issued > 0 && !sig.trim().is_empty() {
```
Comment (logicminds) [inline src-tauri/src/db.rs]: **P1 #1** — Verify only runs when  and the signature is non-empty, so this command still persists unsigned roster rows. MLS v2 ingest is stricter, so Crew can show an uncertified address that peers will not accept.

Require  and a passing  before . Add a test that  returns Err.

File: src-tauri/src/evm/nave_pirata_deploy.rs [hunk 2]
```diff
@@ -334,7 +334,13 @@ pub(crate) fn ensure_captain_for_parent_deploy<R: Runtime>(
         if bound_squad_address_for_parent(app, pid)?.as_ref() == Some(&captain) {
             let hex = format!("{captain:#x}");
             for heal_pid in &candidates {
-                db::upsert_squad_member_evm(app.clone(), (*heal_pid).to_string(), hex.clone())?;
+                db::upsert_squad_member_evm(
+                    app.clone(),
+                    (*heal_pid).to_string(),
+                    hex.clone(),
+                    None,
+                    None,
```
Comment (logicminds) [inline src-tauri/src/evm/nave_pirata_deploy.rs]: **P1 #1 (call site)** — Captain deploy heal writes the shared roster with / unset, which becomes  /  in .

Sign a bind cert here, or skip writing  and rely on the  binding (what ACL already uses).

**Review by logicminds — CHANGES_REQUESTED** [review]
## Code Review Results

**Scope:** merge-base `85371e2567b5bbb75b8ef595ea19faf9e11bdc81` (`main`) -> `ee9ca5fa6df710c850db637d81c3ccc08ba60311` on `feat/squad-acl` (106 files, +3123/-1781). PR #342, local-aligned.
**Intent:** Prove npub-to-roster-EVM ownership with an off-chain EIP-712 `SquadRosterBind` cert; persist v2 MLS shares and roster snapshots only after verify. Cert is display/gossip only; ACL still uses the `#personal-alerts` binding. Also rename squad bot -> Join inbox and fix inbox lifecycle bugs.
**Mode:** markdown, report-only (no `apply:local`). Findings posted on PR #342.

**Reviewers:** correctness, project-standards, testing, maintainability, security, api-contract, data-migration, reliability, adversarial
- project-standards -- root `AGENTS.md` / `CLAUDE.md` govern the changed paths
- testing -- cert/join-inbox tests plus ingest and wire-v2 behavior
- maintainability -- join-inbox rename plus new cert module (~3124 executable lines)
- security -- npub<->EVM binding, EIP-712 verify, MLS gossip persistence
- api-contract -- `squad_member_evm_share` v2, snapshot wire, Tauri commands
- data-migration -- bind-cert columns and join-inbox tables
- reliability -- join-inbox lifecycle and silent ingest failures
- adversarial -- identity writes and gossip (in-process; no peer CLI)

### P1 -- High

| # | File | Issue | Reviewer | Confidence |
|---|------|-------|----------|------------|
| 1 | `src-tauri/src/db.rs:4120` | Local roster upsert still persists unsigned bind rows | correctness, security, adversarial | 75 |

- **#1** -- MLS v2 ingest requires a verified cert, but `upsert_squad_member_evm` only verifies when `issued_at > 0` and the signature is non-empty. `ensure_captain_for_parent_deploy` still calls it with `None, None`, writing `issued_at = 0` and an empty `bind_signature`. Crew can show an uncertified address while peers refuse v1/unsigned gossip. Require a passing cert on that command, and either sign the deploy heal or stop writing `squad_member_evm` from it (binding table is enough for ACL).

### Actionable Findings

| # | File | Issue | Route | Notes |
|---|------|-------|-------|-------|
| 1 | `src-tauri/src/db.rs:4120` | Unsigned local upsert / deploy heal | `gated_auto -> downstream-resolver` | `suggested_fix` present; add a failing test for `None, None` |

### Coverage

- Fast pass: no urgent P0/P1 candidates from a first-principles scan (unsigned upsert confirmed in the deep pass).
- Cross-model adversarial: **not run** -- host attested as Claude-family (`CLAUDECODE=1`) but no `codex`/`claude`/`grok`/`cursor-agent`/`composer` CLI is installed. In-process adversarial lens ran in the same session after Task spawn returned `401 User not found`.
- Validator batch: **degraded** -- independent validator spawn failed with the same 401. P1 #1 kept after orchestrator re-read of `upsert_squad_member_evm` (db.rs:4120) and `ensure_captain_for_parent_deploy` (nave_pirata_deploy.rs:337-342).
- Mechanical helper: 3 findings; testing-only unsigned-path coverage and single-reviewer future-`issued_at` lock demoted (mode-aware). Same-session personas do not count as independent corroboration; helper promotion to confidence 100 was not used in this report.
- Mode-aware demotions: 2 (testing coverage -> Testing Gaps; adversarial future issued_at -> Residual Risks).
- Intent confidence: **explicit** (PR #342 body).
- Untracked excluded: `.agents/security-review-prompt.md`.
- Subagent dispatch: all 9 Task workers failed immediately (`401 User not found`); review completed on the degraded in-process path.

### Residual Risks

- EIP-712 domain uses `chainId = 0` and `verifyingContract = 0`; certs are chain-replayable. Documented as display/gossip only; ACL still reads `squad_member_evm_account`.
- `WHERE excluded.issued_at >= squad_member_evm.issued_at` with no max-skew check lets a far-future `issued_at` freeze that npub's display row on every peer that ingested it.
- `try_apply_squad_member_evm_share` / snapshot apply swallow verify/upsert errors (`let _ = ...`).
- `publishSquadMemberEvmShare` can succeed on MLS send then fail local upsert, so peers have a cert the publisher DB lacks.
- `may_mint_join_inbox` allows any member when MLS `creator_pubkey` is missing/unparseable (covered by unit tests as intentional).
- Join-inbox SQL is created both in `join_inbox.rs` and `V20260824050229__join_inbox_tables.sql`.
- Existing `squad_member_evm` rows backfill to unsigned (`issued_at=0`, empty signature) via NOT NULL defaults.

### Testing Gaps

- No test that `upsert_squad_member_evm` without cert returns Err, or that captain deploy heal does not insert an empty `bind_signature`.
- No test that far-future `issued_at` is rejected.
- No test that an MLS `squad_evm_roster_snapshot` from a non-member is ignored (Rust has `is_author_mls_member_for_chat`; not asserted from TS).

### Verdict

**Ready with fixes.** Certified MLS ingest (author == member_npub, recover == address, stale issued_at skip) and ACL-from-binding match the PR. Fix #1 before merge so local persist matches the "reject unsigned" contract the PR claims.

File: src-tauri/src/db.rs (hunk 1 above)
Comment (logicminds) [inline src-tauri/src/db.rs]: **P1 #1** — Verify only runs when issued_at > 0 and the signature is non-empty, so this command still persists unsigned roster rows. MLS v2 ingest is stricter, so Crew can show an uncertified address that peers will not accept.

Require issued_at > 0 and a passing verify_squad_roster_bind_cert before upsert. Add a test that upsert_squad_member_evm(..., None, None) returns Err.

File: src-tauri/src/evm/nave_pirata_deploy.rs (hunk 2 above)
Comment (logicminds) [inline src-tauri/src/evm/nave_pirata_deploy.rs]: **P1 #1 (call site)** — Captain deploy heal writes the shared roster with issued_at/bind_signature unset, which becomes 0 / empty in upsert_squad_member_evm.

Sign a bind cert here, or skip writing squad_member_evm and rely on the personal-alerts binding (what ACL already uses).

**Review by logicminds — APPROVED** [review]
